src/test/java/teammates/test/cases/webapi/GetAuthInfoActionTest.java

```diff
@@ -125,9 +130,66 @@ protected void testExecute() {
         assertThrows(UnauthorizedAccessException.class, () -> getAction(new String[] {
                 Const.ParamsNames.USER_ID, "idOfAnotherInstructor",
         }));
+    }
+
+    @Test
+    public void testExecute_addCsrfTokenCookies_shouldAddToResponseAccordingToExistingCsrfToken() {
+
+        ______TS("Test will add CSRF token cookies: No logged in user");
+
+        gaeSimulation.logoutUser();
+
+        MockHttpServletRequest mockRequest = new MockHttpServletRequest(getRequestMethod(),
+                Const.ResourceURIs.URI_PREFIX + getActionUri());
+        mockRequest.addCookie(new Cookie(Const.CsrfConfig.TOKEN_COOKIE_NAME, "some random csrf token"));
```

Comment:
I believe u have done some research on CRSF token.

Here are some typical scenario of the CRSF token.

- User visits the website first time -> CSRF token missing -> CSRF mismatched -> regenerate CSRF token
- User visits the website with existing CSRF token -> CSRF token matched -> do nothing
- User visits the website with expired CSRF token -> CSRF token mismatched -> regenerate CSRF token
- User visits the website with fake CSRF token -> CSRF mismatched -> regenerate CSRF token

Therefore, It is not really that CSRF token is "random".